File: restaurant-client/src/main/java/com/jdc/restaurant/client/utils/RequestInterceptor.java
package com.jdc.restaurant.client.utils;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class RequestInterceptor implements Interceptor{

	@Override
	public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        Request newReq = null;

        try {

            Request.Builder builder = request.newBuilder();

            String token = SessionContext.getToken();
            
            if(null != token) {
                builder = builder.addHeader("Authorization", token);
            }

            newReq = builder.build();

        } catch (Exception e) {
            e.printStackTrace();
        }

        return chain.proceed(null != newReq ? newReq : request);
	}

}

File: restaurant-mobile/app/src/main/java/com/jdc/restaurant/api/utils/RequestInterceptor.kt
package com.jdc.restaurant.api.utils

import android.util.Log
import com.jdc.restaurant.api.ClientContext
import okhttp3.Interceptor
import okhttp3.Response

object RequestInterceptor: Interceptor {

    override fun intercept(chain: Interceptor.Chain): Response {

        val originalRequest = chain.request()
        val newRequest = ClientContext.token?.let {
            Log.d("com.jdc.restaurant.app", ClientContext.token)
            originalRequest.newBuilder().addHeader("Authorization", it).build()
        }

        return chain.proceed(newRequest ?: originalRequest)
    }

}